src/type/Type.php

```diff
@@ -90,4 +95,9 @@ public function isDefined(): bool
     {
         return $this->id !== null || $this->name !== null;
     }
+
+    public static function anyId($name): TypeInterface
+    {
+        return new self(self::ANY, $name);
```

Comment:
```suggestion
        return new static(self::ANY, $name);
```

Otherwise the extended class will not be able to use this method correctly.